ReactAndroid/src/main/java/com/facebook/react/animated/NativeAnimatedNodesManager.java

```diff
@@ -353,9 +355,47 @@ public void onEventDispatch(Event event) {
    */
   public void runUpdates(long frameTimeNanos) {
     UiThreadUtil.assertOnUiThread();
+    boolean hasFinishedAnimations = false;
+
+    List<AnimatedNode> nodes = new ArrayList<AnimatedNode>(mUpdatedNodes.size() + mActiveAnimations.size());
```

Comment:
Re-allocating a new list on every frame isn't great: you can just keep a static one around to re-use.